src/services/preference/preference.service.ts

```diff
@@ -40,22 +79,34 @@ export class PreferenceService {
     // make sure the loaded preference object conforms to type T
     const keys = Object.keys(loadedPreference) as (keyof T)[];
     for (const key of keys) {
-      if (!(key in defaultPreference)) {
+      if (!(key in preferenceMeta)) {
         delete loadedPreference[key];
         continue;
       }
 
-      const expectedType = typeof defaultPreference[key];
+      const expectedMeta = preferenceMeta[key];
       const actualType = typeof loadedPreference[key];
 
       // in case of type mismatch, override with the default value
```

Comment:
What do you mean? I think this comment is relevant to the code below.

Reply:
I believe the comment above refers to line 98 now

Reply:
It refers to the code between line 91 and 100. The "select" type here can be written as `"option1" | "option2" | ... ` in TypeScript, but there is no easy solution to do this type check in runtime. We need to verify the type to be a string and be one of the options so that we can determine if it is a type mismatch.